### PR TITLE
LPD-1617

### PR DESCRIPTION
**Pull request by cdbm**

### CODE REVIEW

**Comment by cdbm** [comment]
ci:test:sf

**Comment by stian-sigvartsen** [comment]
ci:test:sf

**Comment by istvansajtos** [comment]
Just started reviewing. :)

**Comment by istvansajtos** [comment]
Hi @cdbm ,

Sorry for the delayed review. I've commented on your changes, please check them and let me know if you want to talk about it. Also, please check my suggestions below:

1. For the new language property, please regenerate all language files. I think you should run "gradlew buildLang" from portal-language-lang module (or maybe and "build-lang", I'm not exactly sure).
2. For the commits like "Merge branch 'master' into LPD-1617", please avoid them as those would not be accepted eventually. Instead, please rebase your changes on latest master state. 
3. The same goes for "LPD-1617 sf". SF changes should not go into separate commits if possible. Please inline the SF changes into the commits where they actually belong to.

Feel free to contact me if I can help you with anything. Please re-send the PR when you implemented the requested changes.

Thank you!

File: modules/apps/login/login-web/src/main/java/com/liferay/login/web/internal/portlet/action/LoginMVCActionCommand.java [hunk 1]
```diff
@@ -121,6 +133,41 @@ protected void doProcessAction(
...
+				String emailParam = "emailPasswordSent";
+
+				if (company.isSendPasswordResetLink()) {
+					emailParam = "emailPasswordReset";
+				}
+
+				String subject = portletPreferences.getValue(
+					emailParam + "Subject_" + languageId, null);
+				String body = portletPreferences.getValue(
+					emailParam + "Body_" + languageId, null);
```
Comment (istvansajtos) [inline modules/apps/login/login-web/src/main/java/com/liferay/login/web/internal/portlet/action/LoginMVCActionCommand.java]: We need to send out the lockout notification email, we should load those templates instead from portal.properties.

**Comment by istvansajtos** [comment]
Thanks for updating the PR, @cdbm! Functionally your changes look good, I got the expected results when manual testing. 

Regarding my previous concerns, I still see some of them unresolved.
- I don't see the Language*.properteis files regenerated, please let me know if you need help with that.
- The inconsistency of not having null-check on portletPreferenes [here](https://github.com/liferay-appsec/liferay-portal/pull/1622/files#diff-a82a9a2ee0002373e3ca1670c1e99c66d467a90a72aa7ae991e81fde9fcf9b3aR141-R147) but having null-check [here](https://github.com/liferay-appsec/liferay-portal/pull/1622/files#diff-a82a9a2ee0002373e3ca1670c1e99c66d467a90a72aa7ae991e81fde9fcf9b3aR293-R297) should be also clarified.

Regarding the "[LPD-1617 add required changes](https://github.com/liferay-appsec/liferay-portal/pull/1622/commits/b054c8a26bf0dc66b57b67c4de6ec99ccce46e94)" commit, we should avoid committing unrelated changes under a generic commit message for transparency reasons. May I ask you to get rid of this commit and instead fixup each changes into the original commit where it actually belongs to?

Furthermore, just to mention, having a separate semver commit helps when the solution needs to be backported to older branches. Semver changes committed together with logical code changes may cause headache as the almost always collide when cherry-picked.

Thank you!

**Comment by istvansajtos** [comment]
@cdbm , besides the above, I observed semver changes in portal-impl and portal-kernel. Please double-check and commit those as well if necessary. Thank you!

**Comment by istvansajtos** [comment]
ci:test:sf

**Comment by istvansajtos** [comment]
ci:test:relevant

**Comment by istvansajtos** [comment]
ci:test:security

**Comment by cdbm** [comment]
Hi @istvansajtos i revised and updated the pr with the required changes, as of semver changes i run the commands here and the versions remained the same

**Comment by istvansajtos** [comment]
ci:test:sf

**Comment by istvansajtos** [comment]
ci:test:relevant

**Comment by istvansajtos** [comment]
ci:test:security

**Comment by cdbm** [comment]
@istvansajtos as far as i checked ci test errors seem unrelated

**Comment by istvansajtos** [comment]
Hi @cdbm ,

I sent an updated PR, please see https://github.com/liferay-appsec/liferay-portal/pull/1680

FYI, the changes are the following:
- I split you last commit ([LPD-1617 only send email when theres no reset link](https://github.com/liferay-appsec/liferay-portal/pull/1622/commits/ccfcd33e6d1ca58b982b47c09dbf6d4619bd3179)) into logically separated parts and fixup into your first three commits where they actually belong to. Now the first three commits contains all of your changes. I couldn't find a better way to reach transparency. 
- I noticed that point #2 of the criteria documented by Zsigmond is not satisfied completely.
- Semver changes were required. Building the portal locally automatically raised the versions for me.

**Pull request closed without merging**